jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java

```diff
@@ -704,7 +724,193 @@ protected void deployScriptsToUserJwalaScriptsDir(Jvm jvm, User user) throws Com
         }
 
     }
+    
+    public Jvm upgradeJDKAndDeployJvm(String jvmName, User user) {
+        boolean didSucceed = false;
+        Jvm jvm = getJvm(jvmName);
+        LOGGER.debug("Start upgradeJDKAndDeployJvm for {} by user {}", jvmName, user.getId());
+
+        historyFacadeService.write(jvm.getHostName(), jvm.getGroups(), "Starting to upgrade JDK :: JVM " +
+                jvm.getJvmName() + " on host " + jvm.getHostName(), EventType.USER_ACTION_INFO, user.getId());
+
+        //add write lock for multiple write
+        binaryDistributionLockManager.writeLock(jvmName + "-" + jvm.getId().toString());
+
+        try {
+            if (jvm.getState().isStartedState()) {
+                final String errorMessage = "The remote JVM " + jvm.getJvmName() + " must be stopped before attempting to generate the JVM";
+                LOGGER.info(errorMessage);
+                throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE, errorMessage);
+            }
+            //Does this required??
+            validateJvmAndAppResources(jvm);
+
+            checkForJvmBinaries(jvm);
+
+            distributeBinaries(jvm);
+            
+            //This will update the Setenv file with upgraded JDK
+            deployJvmResourceFilesForJDKUpgrade(jvm, user);
+
+
+            // set the state to stopped
+            updateState(jvm.getId(), JvmState.JVM_STOPPED);
+
+            didSucceed = true;
+        } catch (CommandFailureException | IOException e) {
+            LOGGER.error("Failed to upgrade JDK :: the JVM config for {}", jvm.getJvmName(), e);
+            throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE, "Failed to generate the JVM config: " + jvm.getJvmName(), e);
+        } finally {
+            binaryDistributionLockManager.writeUnlock(jvmName + "-" + jvm.getId().toString());
+            LOGGER.debug("End upgrade JDK for {} by user {}", jvmName, user.getId());

```

Comment:
Make this LOGGER.info so we can see that end of the upgrade in the log.

Reply:
This is fixed in the latest check-in.